### ISSUE TITLE
Crawl failing with syntax error

### ISSUE THREAD

**Issue by kantenkugel** [geelen/react-snapshot#16]
I seem to have run into a error which I can't pinpoint the source of. I'm thinking that its some dependency issue.

My setup:
Using create-react-app (react-scripts 0.9.5)
Which uses jsdom 9.12.0, react and react-dom 15.4.2

Having this very basic example:
```js
import React from 'react';
import {render} from 'react-snapshot';

const TestComponent = () => (
    <div><h1>Test</h1></div>
);

render(<TestComponent/>, document.querySelector('main'));
```

Which throws following error when running `react-scripts build && react-snapshot`:
(paths shortened to start at project root)

> Starting crawling http://localhost:2999
> Error: Uncaught [SyntaxError: Unexpected token <]
>     at reportException (...\node_modules\jsdom\lib\jsdom\living\helpers\runtime-script-errors.js:58:24)
>     at processJavaScript (...\node_modules\jsdom\lib\jsdom\living\nodes\HTMLScriptElement-impl.js:130:7)
>     at HTMLScriptElementImpl._eval (...\node_modules\jsdom\lib\jsdom\living\nodes\HTMLScriptElement-impl.js:65:7)
>     at e (...\node_modules\jsdom\lib\jsdom\browser\resource-loader.js:31:22)
>     at Object.check (...\node_modules\jsdom\lib\jsdom\living\nodes\Document-impl.js:89:11)
>     at ...\node_modules\jsdom\lib\jsdom\living\nodes\Document-impl.js:108:12
>     at wrappedEnqueued (...\node_modules\jsdom\lib\jsdom\browser\resource-loader.js:255:16)
>     at Request.request [as _callback] (...\node_modules\jsdom\lib\jsdom\browser\resource-loader.js:203:9)
>     at Request.self.callback (...\node_modules\request\request.js:188:22)
>     at emitTwo (events.js:106:13) SyntaxError: Unexpected token <
>     at Object.exports.runInContext (vm.js:66:16)
>     at processJavaScript (...\node_modules\jsdom\lib\jsdom\living\nodes\HTMLScriptElement-impl.js:128:10)
>     at HTMLScriptElementImpl._eval (...\node_modules\jsdom\lib\jsdom\living\nodes\HTMLScriptElement-impl.js:65:7)
>     at e (...\node_modules\jsdom\lib\jsdom\browser\resource-loader.js:31:22)
>     at Object.check (...\node_modules\jsdom\lib\jsdom\living\nodes\Document-impl.js:89:11)
>     at ...\node_modules\jsdom\lib\jsdom\living\nodes\Document-impl.js:108:12
>     at wrappedEnqueued (...\node_modules\jsdom\lib\jsdom\browser\resource-loader.js:255:16)
>     at Request.request [as _callback] (...\node_modules\jsdom\lib\jsdom\browser\resource-loader.js:203:9)
>     at Request.self.callback (...\node_modules\request\request.js:188:22)
>     at emitTwo (events.js:106:13)
> ✏️   Saving / as /index.html
> �   Finished crawling.

**Comment by kantenkugel** [comment]
I figured out that this problem has to do with the "homepage" field in the package.json file. If the field is not present, it renders as normal, but with homepage field present, it fails with above error.

As I need the homepage field for my builds (react page hosted under sub-folder of domain), it would be nice if this could be fixed

**Comment by kantenkugel** [comment]
I have written a quick proof of concept that works for me (using the public-path): https://gist.github.com/kantenkugel/aee4d827d51e43ff9f1feb3709c0830b

**Comment by Cooke** [comment]
I have also run into this limitation. If you (@kantenkugel) have a solution can you not create a PR?

**Comment by kantenkugel** [comment]
that above PR from kevin looks much better than my ugly code and I also did a lot of other stuff for my personal use so i didn't want to make a PR out of that directly. Wanted to see what the opinion on it is first.

**Comment by kantenkugel** [comment]
Closing issue as this is now implemented in 1.0.5-0. Would be nice to publish to npm tho as the npm registry still has 1.0.4 as its current version
